src/query/util/logging/log.go

```diff
@@ -127,8 +128,14 @@ func withResponseTimeLoggingFunc(
 	return func(w http.ResponseWriter, r *http.Request) {
 		startTime := time.Now()
 		rqCtx := NewContextWithGeneratedID(r.Context())
+		rqID := ReadContextID(rqCtx)
```

Comment:
nit: Move this into the `if sp != nil {..` block since it's not used if sp is nil?